events/2023-google-summer-of-code/index.markdown

```diff
@@ -0,0 +1,67 @@
+---
+author: Mark Moll
+comments: false
+date: 2023-02-02 00:00:00 -0700
+layout: page
+slug: 2023-google-summer-of-code
+title: 2023 Google Summer of Code
+---
+# 2023 Google Summer of Code
+
+MoveIt is planning to participate again in the [Google Summer of Code](https://summerofcode.withgoogle.com). Below is a tentative list of project ideas. If you have your own project idea, feel free to create a pull request on the [repository for this page](https://github.com/ros-planning/moveit.ros.org). Unlike previous years, there are now two different project sizes. Medium projects are 175 hours, while large projects are 350 hours. See the [Google Summer of Code FAQ](https://developers.google.com/open-source/gsoc/faq) for details.
+
+For general MoveIt GSoC questions you can contact [Mark Moll](mailto:mark@picknik.ai) (**but make sure to read the [tips at the bottom of this page first](#tips-for-writing-a-successful-google-summer-of-code-application-for-moveit)**).
+
+## Project ideas
+
+### Improved benchmarking support in MoveIt
+
+* Utilize the [MoveIt Benchmark Suite](https://github.com/captain-yoshi/moveit_benchmark_suite) developed during a previous GSoC project to prepare a set of realistic scenarios to benchmark independent components of the MoveIt eco-system. Based on these tests, we would like to setup a CI runner to evaluate the performance impact of new pull-requests in MoveIt. An advanced component in this might evaluate the benchmark results for statistically significant changes and point these out in a CI report.
+* Expected outcome: New pull requests are automatically checked for performance regressions or improvements.
+* Project size: large (350 hours)
+* Difficulty: medium
+* Preferred skills: C++, Python, MoveIt, Motion Planning, Benchmarking, and Building Infrastructure
+* Mentor: [Michael Görner](mailto:goerner@informatik.uni-hamburg.de)
```

Comment:
I guess this was accomplished last year?
```suggestion
```

Reply:
No, it would be great if it were, but there is no CI integration for the current system and prominently no database of motion plan requests with typical planning scenes that would be benchmarked and analyzed automatically.

Reply:
@v4hn Ah ok, I was not aware of that. Do you think it would make sense to update the project description such that it is a follow-up project?

Reply:
I think it's clear from the first sentence that this is based on a previous GSoC project, no?